apps/chat/src/pages/models/[modelId].tsx

```diff
@@ -0,0 +1,242 @@
+  const dispatch = useAppDispatch();
+
+  const isOverlay = useAppSelector(SettingsSelectors.selectIsOverlay);
+  const shouldLogin = useAppSelector(AuthSelectors.selectIsShouldLogin);
+  const authStatus = useAppSelector(AuthSelectors.selectStatus);
+  const modelIsLoaded = useAppSelector(ModelsSelectors.selectIsModelsLoaded);
+  const activeModel = useAppSelector((state) =>
+    ModelsSelectors.selectModel(state, modelId),
+  );
+  const shouldOverlayLogin = isOverlay && shouldLogin;
+
+  // EFFECTS  --------------------------------------------
+  useEffect(() => {
+    if (!isOverlay && shouldLogin) {
+      signIn();
+    }
+  }, [isOverlay, shouldLogin]);
+
+  useEffect(() => {
+    dispatch(AuthActions.setSession(session));
+  }, [dispatch, session]);
+
+  // ON LOAD --------------------------------------------
+
+  useEffect(() => {
+    // Hack for ios 100vh issue
+    const handleSetProperVHPoints = () => {
+      document.documentElement.style.setProperty(
+        '--vh',
+        window.innerHeight * 0.01 + 'px',
+      );
+      dispatch(UIActions.resize());
+    };
+    handleSetProperVHPoints();
+    window.addEventListener('resize', handleSetProperVHPoints);
+
+    dispatch(SettingsActions.initApp());
+  }, [dispatch, initialState]);
+
+  const handleOverlayAuth = async () => {
+    const timeout = 30 * 1000;
+    let complete = false;
+    await Promise.race([
+      timeoutAsync(timeout),
+      (async () => {
+        const authWindowLocation = new AuthWindowLocationLike(
+          `api/auth/signin`,
+        );
+
+        await authWindowLocation.ready; // ready after redirects
+        const t = Math.max(100, timeout / 1000);
+        // wait for redirection to back
+        while (!complete) {
+          try {
+            if (authWindowLocation.href === window.location.href) {
+              complete = true;
+              authWindowLocation.destroy();
+              break;
+            }
+          } catch {
+            // Do nothing
+          }
+          await delay(t);
+        }
+        window.location.reload();
+
+        return;
+      })(),
+    ]);
+  };
+
+  return (
+    <>
+      <Head>
+        <title className="whitespace-pre">{activeModel?.name || modelId}</title>
+        <meta name="description" content="ChatGPT but better." />
+        <meta
+          name="viewport"
+          content="height=device-height ,width=device-width, initial-scale=1, user-scalable=no"
+        />
+      </Head>
+
+      {shouldOverlayLogin ? (
+        <div className="grid size-full min-h-[100px] place-items-center bg-layer-1 text-sm text-primary">
+          <button
+            onClick={handleOverlayAuth}
+            className="button button-secondary"
+            disabled={authStatus === 'loading'}
+          >
+            {t('Login')}
+          </button>
+        </div>
+      ) : (
+        <>
+          {modelId && modelIsLoaded && !activeModel ? (
+            <div className="pt-2">
```

Comment:
```suggestion
            <div className="h-screen pt-2">
```
->
![image](https://github.com/epam/ai-dial-chat/assets/20163971/063c9af9-31fd-46db-b030-8844d1550cd4)

Reply:
Done